eclipse-scout-core/src/form/FormController.js

```diff
@@ -221,9 +221,10 @@ export default class FormController {
     }
   }
 
-  _findFormToActivateAfterDialogRemove() {
-    if (this.displayParent.dialogs.length > 0) {
-      return this.displayParent.dialogs[this.displayParent.dialogs.length - 1];
+  _findFormToActivateAfterDialogRemove(removedDialog) {
+    const dialogs = this.displayParent.dialogs.filter(d => d !== removedDialog);
+    if (dialogs > 0) {
```

Comment:
this should probably be dialogs.length > 0. It is interesting, that no spec fails. At least "will be set to the display parent form if dialog closes" should fail, but maybe there is a different logic that handles that case.